babel/src/test/java/org/apache/calcite/test/BabelParserTest.java

```diff
@@ -340,6 +340,56 @@ private void checkParseInfixCast(String sqlType) {
     sql(sql).ok(expected);
   }
 
+  @Test public void testTableAttributeIsolatedLoadingDefault() {
+    final String sql = "create table foo, with isolated loading (bar integer)";
+    final String expected = "CREATE TABLE `FOO`, WITH ISOLATED LOADING (`BAR` INTEGER)";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testTableAttributeIsolatedLoadingNoModifier() {
+    final String sql = "create table foo, with no isolated loading (bar integer)";
+    final String expected = "CREATE TABLE `FOO`, WITH NO ISOLATED LOADING (`BAR` INTEGER)";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testTableAttributeIsolatedLoadingConcurrentModifier() {
+    final String sql = "create table foo, with concurrent isolated loading (bar integer)";
+    final String expected = "CREATE TABLE `FOO`, WITH CONCURRENT ISOLATED LOADING (`BAR` INTEGER)";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testTableAttributeIsolatedLoadingNoConcurrentModifier() {
+    final String sql = "create table foo, with no concurrent isolated loading (bar integer)";
+    final String expected = "CREATE TABLE `FOO`, WITH NO CONCURRENT ISOLATED LOADING "
+        + "(`BAR` INTEGER)";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testTableAttributeIsolatedLoadingForAllOperationLevel() {
+    final String sql = "create table foo, with isolated loading for all (bar integer)";
+    final String expected = "CREATE TABLE `FOO`, WITH ISOLATED LOADING FOR ALL (`BAR` INTEGER)";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testTableAttributeIsolatedLoadingForInsertOperationLevel() {
+    final String sql = "create table foo, with isolated loading for insert (bar integer)";
+    final String expected = "CREATE TABLE `FOO`, WITH ISOLATED LOADING FOR INSERT (`BAR` INTEGER)";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testTableAttributeIsolatedLoadingForNoneOperationLevel() {
+    final String sql = "create table foo, with isolated loading for none (bar integer)";
+    final String expected = "CREATE TABLE `FOO`, WITH ISOLATED LOADING FOR NONE (`BAR` INTEGER)";
+    sql(sql).ok(expected);
+  }
+
+  @Test public void testTableAttributeIsolatedLoadingModifierOperationLevel() {
+    final String sql = "create table foo, with concurrent isolated loading for none (bar integer)";
+    final String expected = "CREATE TABLE `FOO`, WITH CONCURRENT ISOLATED LOADING FOR NONE "
+        + "(`BAR` INTEGER)";
+    sql(sql).ok(expected);
+  }
+
```

Comment:
Combining all the options with a "WITH NO CONCURRENT ISOLATED LOADING FOR ALL" test could be good too. Might be overkill, up to you.

Reply:
I think the last test there might be enough since we have "no" and "concurrent" already together in a test, and another test with a modifier and the "for" keyword together